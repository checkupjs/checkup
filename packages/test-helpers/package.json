{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
