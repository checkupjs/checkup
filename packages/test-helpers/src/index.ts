export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
