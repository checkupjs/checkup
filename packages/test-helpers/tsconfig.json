{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
