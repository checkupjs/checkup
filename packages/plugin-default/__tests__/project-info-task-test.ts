import { CheckupProject, stdout } from '@checkup/test-helpers';
import ProjectInfoTask from '../src/tasks/project-info-task';
import ProjectInfoTaskResult from '../src/results/project-info-task-result';

describe('project-info-task', () => {
  let checkupProject: CheckupProject;

  describe('for Projects', () => {
    beforeEach(() => {
      checkupProject = new CheckupProject('checkup-app', '0.0.0', project => {
        project.addDependency('ember-cli', '^3.15.0');
      });

      checkupProject.writeSync();
      checkupProject.gitInit();
    });

    afterEach(() => {
      checkupProject.dispose();
    });

    it('can read project info and output to console', async () => {
      const result = await new ProjectInfoTask({ path: checkupProject.baseDir }).run();
      const taskResult = <ProjectInfoTaskResult>result;

      taskResult.stdout();

      expect(stdout()).toMatchSnapshot();
    });

    it('can read project info as JSON', async () => {
      const result = await new ProjectInfoTask({ path: checkupProject.baseDir }).run();
      const taskResult = <ProjectInfoTaskResult>result;

      expect(taskResult.json()).toMatchSnapshot();
    });
  });
});
