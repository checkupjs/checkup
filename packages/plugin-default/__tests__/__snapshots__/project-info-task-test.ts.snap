// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-info-task for Projects can read project info and output to console 1`] = `
"=== Project Information

name:    checkup-app
version: 0.0.0

=== Repository Information
Active days:   0 days
Age:           0 days
Total commits: 0
Total files:   0

"
`;

exports[`project-info-task for Projects can read project info as JSON 1`] = `
Object {
  "project-info": Object {
    "name": "checkup-app",
    "repository": Object {
      "activeDays": "0 days",
      "age": "0 days",
      "totalCommits": 0,
      "totalFiles": 0,
    },
    "version": "0.0.0",
  },
}
`;
